lib/routers/rops.js

```diff
@@ -0,0 +1,53 @@
+const { Router } = require('express');
+const { BadRequestError } = require('@asl/service/errors');
+
+module.exports = settings => {
+  const router = Router({ mergeParams: true });
+
+  router.get('/', (req, res, next) => {
+    if (!req.query.year) {
+      throw new BadRequestError('year must be provided');
+    }
+
+    const year = req.query.year;
+    let due = 0;
+    let submitted = 0;
+    let outstanding = 0;
+
+    Promise.resolve()
+      .then(() => {
+        const q = req.db.asl('projects')
+          .select('id')
```

Comment:
Is there a strong reason not to do a `.count()` here?

Reply:
Not really, knex just returns counts in an awkward way, e.g. `[ { count: '37' } ]` so I can either do it this way or a `parseInt(projects[0], 10)` because I need to do a subtraction on the result. It's only returning the id for each row anyway so performance-wise it probably makes very little difference.

Reply:
FWIW, it's postgres rather than knex

https://github.com/brianc/node-pg-types#use

Reply:
I would be very ok with doing the thing they describe in that link to fix the issue in our project btw.

Reply:
In fact, [we already do most of that for timestamps](https://github.com/UKHomeOffice/asl-schema/blob/master/index.js#L5-L22), so it's a JFDI I think.

Reply:
I'll do this in a separate schema PR just cause it might affect more stuff.

Reply:
see https://github.com/UKHomeOffice/asl-metrics/pull/76